nimare/tests/test_estimator_performance.py

```diff
@@ -0,0 +1,268 @@
+    elif isinstance(meta, mkda.KDA) and isinstance(corr, FDRCorrector):
+        # ValueError: <class 'nimare.correct.FDRCorrector'> requires "p" maps
+        # to be present in the MetaResult, but none were found.
+        corr_expectation = pytest.raises(ValueError)
+    elif isinstance(meta, mkda.MKDADensity) and corr.method == "bonferroni":
+        # ValueError: <class 'nimare.correct.FWECorrector'> requires "p" maps
+        # to be present in the MetaResult, but none were found.
+        corr_expectation = pytest.raises(ValueError)
+    elif isinstance(meta, mkda.KDA) and corr.method == "montecarlo":
+        # TypeError: correct_fwe_montecarlo() got an unexpected keyword argument 'voxel_thresh'
+        corr_expectation = pytest.raises(TypeError)
+    elif isinstance(meta, mkda.KDA) and corr.method == "bonferroni":
+        # ValueError: <class 'nimare.correct.FWECorrector'> requires "p" maps
+        # to be present in the MetaResult, but none were found.
+        corr_expectation = pytest.raises(ValueError)
+    else:
+        corr_expectation = does_not_raise()
+
+    with corr_expectation:
+        cres = corr.transform(res)
+
+    # if multiple correction failed (expected) do not continue
+    if isinstance(corr_expectation, type(pytest.raises(ValueError))):
+        pytest.xfail("this meta-analysis & corrector combo fails")
+        return 0
+
+    ################################################
+    # CHECK IF P-VALUES ARE WITHIN THE CORRECT RANGE
+    ################################################
+    # default value to assume p-value outputs are correct
+    contains_invalid_p_values = False
+    # mask to include only brain voxels
+    if corr.method == "montecarlo":
+        # the array only contains in-brain voxels
+        logp_values_data = cres.get_map(
+            "logp_level-voxel_corr-FWE_method-montecarlo", return_type="array"
+        )
+        # transform logp values back into regular p values
+        p_values_data = 10 ** -logp_values_data
+
+        if isinstance(meta, mkda.KDA) and isinstance(kern, kernel.MKDAKernel):
+            # KDA with the MKDAKernel gives p-values twice as small as allowed
+            assert p_values_data.min() >= (1.0 / (corr.parameters["n_iters"] * 2))
+        else:
+            # there should not be p-values less than 1 / n_iters
+            assert p_values_data.min() >= (1.0 / corr.parameters["n_iters"])
+
+        # max p-values should be less than 1
+        assert p_values_data.max() < 1.0
+    else:
+        p_values_data = cres.get_map("p", return_type="array")
+
+    # ensure all values are between 0 and 1 (inclusively)
+    assert (p_values_data >= 0).all() and (p_values_data <= 1).all()
+
+    ################################################################
+    # CHECK IF META/KERNEL/CORRECTOR COMBINATION PERFORMS ADEQUATELY
+    ################################################################
+    # The below combinations cannot even detect
```

Comment:
This worries me a bit... I don't see an obvious reason why these combinations should produce markedly different results. @tsalo, am I missing something?

If the issue is that we just happen to have titrated the sampling procedure so that some estimators detect effects but others don't, we should probably change it so that the signal is unambiguous. But it seems unlikely we would just randomly stumble into such a regime, so it's worth looking into what's going on (assuming I'm not misunderstanding something or missing something obvious). If it's just that the monte carlo correction procedure is more conservative, we should probably boost the signal to make sure that we still detect it. But I would still worry if some estimator/kernel combinations "work" with monte carlo correction and others don't.

Reply:
I'd imagine that ALE+MKDAKernel is pretty close to invalid, even if it successfully runs, so I'm not really surprised by any particular results with that combination. I'm also not surprised by the Bonferroni results of the MKDA+KDAKernel or MKDA+MKDAKernel, but I _am_ surprised that those two don't find anything with FDR correction...

Reply:
BTW I think the Monte Carlo procedure is _less_ conservative (at least with 100 iterations), if I'm reading this section correctly.